core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c

```diff
@@ -2943,3 +3117,88 @@ fd_prestats_destroy(struct fd_prestats *pt)
         wasm_runtime_free(pt->prestats);
     }
 }
+
+bool
+addr_pool_init(struct addr_pool *addr_pool)
+{
+    addr_pool->next = NULL;
+    addr_pool->addr = 0;
+    addr_pool->mask = 0;
+    return true;
+}
+
+bool
+addr_pool_insert(struct addr_pool *addr_pool, const char *addr, uint8_t mask)
+{
+    struct addr_pool *cur = addr_pool;
+    struct addr_pool *next;
+
+    if (!addr_pool) {
+        return false;
+    }
+
+    if (!(next = wasm_runtime_malloc(sizeof(struct addr_pool)))) {
+        return false;
+    }
+
+    next->next = NULL;
+    next->mask = mask;
+    if (os_socket_inet_network(addr, &next->addr) != BHT_OK) {
+        wasm_runtime_free(next);
+        return false;
+    }
+
+    /* attach with */
+    while (cur->next) {
+        cur = cur->next;
+    }
+    cur->next = next;
+    return true;
+}
+
+static bool
+compare_address(const struct addr_pool *addr_pool_entry, const char *addr)
+{
+    /* host order */
+    uint32_t target;
+    uint32_t address = addr_pool_entry->addr;
```

Comment:
Had better use uint32 but not uint32_t here, we usually use uint32/uint16 like types inside runtime, and use uint32_t/uint16_t like types in header files of core/iwasm/include

Reply:
Got it.